fhir-persistence-schema/src/main/resources/set_tenant.sql

```diff
@@ -20,26 +20,39 @@ BEGIN
       -- preventing accidental exposure of tenant-specific data
       SET fhir_admin.sv_tenant_id = NULL;
   ELSE
-	  -- set the sv_tenant_id session variable for the 
-	  -- given tenant_name, but only if the correct key
-	  -- is provided
-	  SELECT t.mt_id, t.tenant_status INTO fhir_admin.sv_tenant_id, v_tenant_status
-	    FROM fhir_admin.tenants t
-	   WHERE t.tenant_name = p_tenant_name
-	     AND EXISTS (SELECT 1 FROM fhir_admin.tenant_keys tk WHERE tk.mt_id = t.mt_id
-	     AND tk.tenant_hash = sysibm.hash(tk.tenant_salt || p_tenant_key, 2));
-	
-	  IF v_not_found = 1
-	  THEN
-	    -- applications should not be passing invalid keys, so treat this as
-	    -- an exception
-	    SIGNAL SQLSTATE '99401' SET MESSAGE_TEXT = 'NOT AUTHORIZED: INVALID TENANT ID';
-	  ELSEIF v_tenant_status = 'FROZEN'
-	  THEN
-	    -- the tenant_key is valid, but the status of tenant has been set to frozen
-	    -- which means the tenant data should not be accessible
-	    SIGNAL SQLSTATE '99401' SET MESSAGE_TEXT = 'NOT AUTHORIZED: TENANT IS FROZEN';
-	  END IF;
-  END IF;
+      -- set the sv_tenant_id session variable for the 
+      -- given tenant_name, but only if the correct key
+      -- is provided
+      SELECT t.mt_id, t.tenant_status INTO fhir_admin.sv_tenant_id, v_tenant_status
+      FROM fhir_admin.tenants t
+      WHERE t.tenant_name = p_tenant_name
+            AND EXISTS (SELECT 1 FROM fhir_admin.tenant_keys tk WHERE tk.mt_id = t.mt_id
+            AND tk.tenant_hash = sysibm.hash(tk.tenant_salt || p_tenant_key, 2));
+
+      IF v_not_found = 1
+      THEN
+        -- If v_not_found, we check only one of the conditions above to see if it is valid
+        -- applications should not be passing invalid keys, so treat this as
+        -- an exception
 
-END
+        -- Check the cause... bad tenant_name
+        SELECT -1 INTO v_not_found
+        FROM fhir_admin.tenants t
+            WHERE t.tenant_name = p_tenant_name;
```

Comment:
sorry, what does SELECT -1 INTO v_not_found do?

Reply:
I guess it only sets if WHERE selects one or more rows?

Reply:
this is to find out the cause is either tenant_name or other. 
this allows for a differentiated signal back from the stored procedure. 

Yes, that's right